static/src/javascripts/projects/facia/modules/onwards/weather.js

```diff
@@ -232,61 +227,28 @@ define([
         },
 
         render: function (weatherData, city) {
-            $weather = $('.weather');
-            $holder = $('.js-container--first .js-container__header');
-
-            $weather = $.create(template(weatherTemplate, {
-                location: city,
-                icon: weatherData.weatherIcon,
-                description: weatherData.weatherText,
-                tempNow: this.getTemperature(weatherData)
-            }));
+            this.attachToDOM(weatherData.html, city);
 
-            $weather.appendTo($holder);
             this.bindEvents();
             this.addSearch();
 
             this.render = function (weatherData, city) {
-                $weather = $('.weather .js-weather-current');
-
-                var $weatherIcon = $('.js-weather-icon', $weather);
-
-                $('.js-weather-temp', $weather).text(this.getTemperature(weatherData));
-                searchTool.setInputValue(city);
-
-                // Replace number in weather icon class
-                $weatherIcon.attr('class', $weatherIcon.attr('class').replace(/(\d+)/g,
-                    weatherData.weatherIcon))
-                    .attr('title', weatherData.weatherText);
-
-                // Close editing
-                this.toggleControls();
+                this.attachToDOM(weatherData.html, city);
+                searchTool.bindElements($('.js-search-tool'));
             };
         },
 
+        attachToDOM: function (tmpl, city) {
+            $holder = $('.js-container--first .js-container__header');
+            $holder.empty();
+            $holder.html(tmpl.replace(new RegExp('{{city}}', 'g'), city));
```

Comment:
I would like to do all fastdom updates after this will be deployed as separate PR. There are more places that can use it around weather js.